src/main/java/net/wimpi/modbus/net/TCPConnectionHandler.java

```diff
@@ -70,7 +70,13 @@ public void run() {
                 if (ModbusCoupler.getReference().getProcessImage() == null) {
                     response = request.createExceptionResponse(Modbus.ILLEGAL_FUNCTION_EXCEPTION);
                 } else {
-                    response = request.createResponse();
+                    try {
+                        response = request.createResponse();
+                    }
+                    catch (Exception e) {
+                        logger.error("Error while processing request", e);
```

Comment:
Perhaps we could implement `toString()` methods in all/most subclasses of `ModbusRequest` and include the `request` variable in the logged error message, so that we have a better idea of what was attempted (the modbus function code and register address) and with which data (the register value).

A proper `toString` implementation would make debugging of JAMOD related code also easier for us.

Reply:
Implemented toString() on request and response classes and added request to log statement in case of error.